src/components/manager.ts

```diff
@@ -736,7 +742,12 @@ export class Manager {
         if (['.tex', '.bib'].concat(this.weaveExt).includes(path.extname(file)) &&
             !file.includes('expl3-code.tex')) {
             this.parseFileAndSubs(file, true)
-            this.updateCompleterOnChange(file)
+            this.updateCompleterOnChange(file).then(() => {
+                const configuration = vscode.workspace.getConfiguration('latex-workshop')
+                if (configuration.get('check.duplicatedLabels.run') === 'onSave') {
+                    this.extension.duplicateLabels.run(file)
+                }
+            })
```

Comment:
We should use `async` and `await`.

Reply:
I cannot really see the benefit of using `async` and `await` over `.then`. Can you expand a bit on this?

Reply:
It makes codes more readable.